File: 901-online-stock-span/901-online-stock-span.java
class StockSpanner {
    Stack<int[]> stk;
    public StockSpanner() {
        stk=new Stack<>();
    }
    
    public int next(int price) {
        int ans=1;
        while(!stk.isEmpty() && stk.peek()[0]<=price)
            ans += stk.pop()[1];
        stk.push(new int[]{price, ans});
        return ans;
    }
}

/**
 * Your StockSpanner object will be instantiated and called as such:
 * StockSpanner obj = new StockSpanner();
 * int param_1 = obj.next(price);
 */

File: 901-online-stock-span/901-online-stock-span.py
class StockSpanner:
    def __init__(self):
        self.stk=[]

    def next(self, price: int) -> int:
        ans=1
        while len(self.stk)>0 and self.stk[len(self.stk)-1][0]<=price:
            ans+=self.stk.pop()[1]
        self.stk.append([price, ans])
        return ans


# Your StockSpanner object will be instantiated and called as such:
# obj = StockSpanner()
# param_1 = obj.next(price)